howdju-common/lib/urlTextFragments.ts

```diff
@@ -102,8 +106,7 @@
   options: ExtractQuotationFromTextFragmentOptions = defaultOptions
 ): string | undefined {
   const urlObj = new URL(url);
-  const hash = urlObj.hash.replace(/^#/, "");
-  const fragmentMatch = hash.match(/^:~:(.*)$/);
+  const fragmentMatch = urlObj.hash.match(/:~:(.*)$/);
```

Comment:
I think this may be a false positive. I tried `const match = "::a::a::a::a::a::a".repeat(10000000).match(/:~:(.*)$/)` in node CLI and it completes in less than a second.